### DISCUSSION TITLE
Electron build failed.

### DISCUSSION THREAD

**Discussion by hyy215** [eclipse-theia/theia#13760] [Q&A] [answered]
### Bug Description:
Electron build failed.

### Steps to Reproduce:

1. Update code to latest and build electron
```
git clean -xxfd
   && yarn
   && yarn electron build
```
2. Rebuild failed.

### Additional Information
![image](https://github.com/eclipse-theia/theia/assets/34715914/3772c735-5ba7-4c7d-a7d3-1bbd542451b2)

- Operating System:win10
- Theia Version: master [0941a8f](https://github.com/eclipse-theia/theia/commit/0941a8ffd9cacceb24b4cab2a003beee2ccf75e8)
- node: 18.17.0

**Comment by msujew** [comment]
Since the electron build seems to generally work, I've converted this into a discussion, as we like to keep issues actionable.

Can you provide a bit more information? 

* Does this on every system you have available? I.e. can you try this on another Windows or a Linux/Mac system. 
* Does this only happen on a "fresh" system (i.e. one that never built Theia before) or is the problem relatively new?
* Can you try re-installing node with the *native build tools* option checked?

**Reply by hyy215** [reply]
> Does this on every system you have available? I.e. can you try this on another Windows or a Linux/Mac system.

It succeeds on Mac.
Another windows also failed.

> Does this only happen on a "fresh" system (i.e. one that never built Theia before) or is the problem relatively new?

After electron upgrade to 28.2.10 [chore: update electron to ^28.2.8 #13580](https://github.com/eclipse-theia/theia/pull/13580)，the electron can't build.

> Can you try re-installing node with the native build tools option checked?

Maybe it's an node-gyp issue. [node-gyp fails for rebuilding cpu-features #2983](https://github.com/nodejs/node-gyp/issues/2983)

**Reply by msujew** [reply]
It's a mix of node-gyp and electron-build, yes. See also https://github.com/eclipse-theia/theia/pull/13748. You'll need a similar resolution for `nan` and `cpu-features` to build.

**Reply by jfaltermeier** [reply, marked as answer]
I also saw a similar issue on the Eclipse Foundation's CI infrastructure for WIndows. There updating Microsoft Visual Studio to the latest 2019 version fixed the issue

**Reply by hyy215** [reply]
> I also saw a similar issue on the Eclipse Foundation's CI infrastructure for WIndows. There updating Microsoft Visual Studio to the latest 2019 version fixed the issue

Thank you for your reply.
The issue is solved when updating Microsoft Visual Studio to 2019

**Reply by PhamQuocThai1303** [reply]
but im in 2022 version of Microsoft Visual Studio, i still got the error with node-gyp
